// framework7-vue/components/menu-dropdown.js
import { renderSlot as _renderSlot, normalizeStyle as _normalizeStyle, createElementVNode as _createElementVNode, normalizeClass as _normalizeClass, openBlock as _openBlock, createElementBlock as _createElementBlock } from "vue";

function render(_ctx, _cache) {
  return _openBlock(), _createElementBlock("div", {
    class: _normalizeClass(_ctx.classes)
  }, [_createElementVNode("div", {
    class: "menu-dropdown-content",
    style: _normalizeStyle({
      height: _ctx.contentHeight
    })
  }, [_renderSlot(_ctx.$slots, "default")], 4)], 2);
}

import { computed } from 'vue';
import { classNames } from '../shared/utils.js';
import { colorClasses, colorProps } from '../shared/mixins.js';
export default {
  name: 'f7-menu-dropdown',
  render,
  props: {
    contentHeight: String,
    position: String,
    left: Boolean,
    center: Boolean,
    right: Boolean,
    ...colorProps
  },

  setup(props) {
    const classes = computed(() => {
      let positionComputed = props.position || 'left';
      if (props.left) positionComputed = 'left';
      if (props.center) positionComputed = 'center';
      if (props.right) positionComputed = 'right';
      return classNames('menu-dropdown', `menu-dropdown-${positionComputed}`, colorClasses(props));
    });
    return {
      classes
    };
  }

};
